### PR TITLE
Add GHA workflow for testing with agent

### PR DESCRIPTION
**Pull request by minusworld**

### CODE REVIEW

File: .github/workflows/semgrep-rules-test-develop.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ on:
 jobs:
   # Note: if you change this test there will likely need to be a
   # corresponding change in returntocorp/semgrep
-  test-latest:
+  test-develop:
```
Comment (emjin) [inline .github/workflows/semgrep-rules-test-develop.yml:12]: Why did this get changed? Not blocking, just wondering

**Review by emjin — APPROVED** [review]
Thanks!

**Comment by emjin** [comment]
Has this been/will this be tested? It should be sufficient to try to release, since semgrep-agent is still behind.

File: .github/workflows/semgrep-rules-test-develop.yml (hunk 1 above)
Comment (minusworld) [reply .github/workflows/semgrep-rules-test-develop.yml:12]: Oh, so that it had a different name in the GitHub UI. I forgot that I made this change here.

**Comment by minusworld** [comment]
> Has this been/will this be tested? It should be sufficient to try to release, since semgrep-agent is still behind.

Yes, I tested it on this [commit here](https://github.com/returntocorp/semgrep-rules/pull/1922/commits/9f19d5c0027bcca811d95e79c54bdbe4f5811f81). The failure is expected because the `metavariable-analysis` rules are still in `develop`.

image

**Pull request merged**